protos/flow.proto

```diff
@@ -206,6 +209,10 @@ message SetupNormalizedTableInput {
 message SetupNormalizedTableBatchInput {
   peerdb_peers.Peer peer_connection_config = 1;
   map<string, TableSchema> table_name_schema_mapping = 2;
+
+  // migration related columns
+  string soft_delete_col_name = 4;
```

Comment:
consider using `optional string` instead so we can distinguish between unset and set to empty string easier